go/vt/topotools/routing_rules.go

```diff
@@ -158,4 +158,19 @@ func SaveKeyspaceRoutingRules(ctx context.Context, ts *topo.Server, rules map[st
 	return ts.SaveKeyspaceRoutingRules(ctx, keyspaceRoutingRules)
 }
 
+func SaveKeyspaceRoutingRulesLocked(ctx context.Context, ts *topo.Server, reason string,
```

Comment:
This function and the one before it (`SaveKeyspaceRoutingRules`) are backwards. Typically in Vitess, the one that doesn't have `locked` in the name needs to acquire the lock before calling the `locked` version. For example, take a look at `ReparentShard` and `reparentShardLocked` in planned_reparenter.go. We also typically make the `locked` version private so that it cannot be called from outside the package.
However, is there still ever a use case where we want people to be able to call `SaveKeyspaceRoutingRules` and it does its work without taking a lock? I don't think so. So the right thing will be to change it to take a lock before doing the save. If we do that, I don't see the need to introduce a `Locked` version at all. We just change the existing function to first take a lock and then execute the callback.   In other words, whatever you have implemented as `SaveKeyspaceRoutingRulesLocked` just becomes the new version of `SaveKeyspaceRoutingRules`.